### PR TITLE
2547 make prominent word naming clearer

### PR DESCRIPTION
**Pull request by hansjovis**
## Summary

This PR can be summarized in the following changelog entry:

* Refactors the code to use another term than 'prominent words', 'keyword suggestions' or 'relevant words'.

## Relevant technical choices:

* I boyscouted the `WordOccurrencesWrapper` class in the components app. I added comments and renamed some variables, methods and functions.

## Test instructions

This PR can be tested by following these steps:

**Note**: this PR needs to be tested in unison with the premium PR: https://github.com/Yoast/wordpress-seo-premium/pull/2613

* Check if the readability- and SEO-analysis work as expected.
* Check if the word occurrences insight works as expected.
* Check if the internal linking suggestions work as expected.
* Check if site-wide internal linking works as expected.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes
* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Quality assurance

* [ ] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Fixes #https://github.com/Yoast/wordpress-seo-premium/issues/2547

### CODE REVIEW

File: apps/components/WordOccurrencesWrapper.js [hunk 1]
```diff
@@ -27,136 +28,242 @@ const initialRelevantWords = [
...
 
 		this.state = {
-			relevantWords: cloneWords( initialRelevantWords ),
+			words: cloneWords( initialRelevantWords ),
 		};
-		this.changeRelevantWord = this.changeRelevantWord.bind( this );
-		this.removeRelevantWord = this.removeRelevantWord.bind( this );
-		this.addRelevantWordRow = this.addRelevantWordRow.bind( this );
-		this.resetRelevantWords = this.resetRelevantWords.bind( this );
+		this.change = this.change.bind( this );
```
Comment (maartenleenders) [inline apps/components/WordOccurrencesWrapper.js]: In my opinion this refactor has gone too far in the other direction: `this.change()` is essentially `WordOccurrencesWrapper.change()` which does not tell me what I am doing. `changeWord`,`removeWord`, `addWord` would have been fine for the issue, and more descriptive than this. 

Then again it's just the test app so 🤷‍♀

File: packages/components/src/index.js [hunk 2]
```diff
@@ -43,7 +43,8 @@ export { default as Icon } from "./Icon";
 export { default as IconButtonToggle } from "./IconButtonToggle.js";
 export { default as IFrame } from "./IFrame";
 export { default as Input } from "./input/Input";
-export { default as KeywordSuggestions } from "./KeywordSuggestions";
+export { default as KeywordSuggestions } from "./WordOccurrenceInsights";
```
Comment (nataliashitova) [inline packages/components/src/index.js]: I understand that it is important that apps that make use of `KeywordSuggestions` don't break in this commit. But in a log run, we don't want things like that in our code. 
@atimmer Is it possible to throw a deprecation warning in this situation?

File: packages/components/src/WordOccurrenceInsights.js [hunk 3]
```diff
@@ -64,24 +64,24 @@ const getKeywordSuggestionExplanation = keywords => {
  * @summary WordList component.
  *
  * @param {string}   title           The title of the list.
- * @param {WordCombination[]|ProminentWord[]}   relevantWords   The relevant words.
+ * @param {WordCombination[]|ProminentWord[]}   words   The relevant words.
```
Comment (nataliashitova) [inline packages/components/src/WordOccurrenceInsights.js]:
```suggestion
 * @param {WordCombination[]|ProminentWord[]}   words   The words to display in the list.
```

File: packages/components/src/index.js (hunk 2 above)
Comment (atimmer) [reply packages/components/src/index.js]: It's possible to throw a deprecation warning, but in this case I would:

* Move this line to the bottom of the file with a `// Deprecated:` comment above it.
* Add `export { default as KeywordSuggestions } from "./WordOccurrenceInsights";` as the new way to import the component in the future.
* Add a line to the changelog detailing that `KeywordSuggestions` is deprecated and should be replaced with `WordOccurrenceInsights`.

Comment (hansjovis) [reply packages/components/src/index.js]: I implemented the suggested changes 🙂.

**Pull request merged**